backend/server/controllers/users.js

```diff
@@ -245,5 +245,49 @@ module.exports = {
         res.status(400).send('Error removing assistant.')
       }
     }
+  },
+
+  async removeStudent(req, res) {
+    if (!helper.controllerBeforeAuthCheckAction(req, res)) {
+      return
+    }
+    if (req.authenticated.success) {
+      try {
+        const userId = req.decoded.id
+
+        const siToRemove = await StudentInstance.findOne({
+          raw: true,
+          where: {
+            id: req.body.id
+          }
+        })
+        if (!siToRemove) {
+          return res.status(400).send('The student instance does not exist')
+        }
+
+        const teacher = await TeacherInstance.findOne({
+          where: {
+            userId: req.decoded.id,
+            courseInstanceId: siToRemove.courseInstanceId
+          }
+        })
+        if (siToRemove.userId !== userId && !teacher) {
+          return res.status(400).send('You must be the student himself or the teacher of the courrse to remove this student instance')
```

Comment:
`courrse` => `course`, and full stop missing. This could also be a 403 instead of 400.